Functions/Gherkin.Tests.ps1

```diff
@@ -120,3 +120,118 @@ Describe "Mocking works in Gherkin" -Tag Gherkin {
         @($gherkin.Results.PassedScenarios).Count | Should Be 3
     }
 }
+
+
+InModuleScope "Pester" {
```

Comment:
I noticed two things with this test. The first thing is really simple to try, the second is a bit more work.

1. I noticed that _all_ tests in this file are tagged with `-Tag Gherkin` while this test is not. I'm not sure if there's anything "special" about this tag other than categorization (probably not), so this likely isn't it&mdash;but the test should still probably be tagged regardless, if only for consistency.

2. I'm not sure why what I'm about to suggest would work any better than what you already have, but, all of the other Gherkin tests spawn a job into which they "re-load" the Pester module so that they don't "monkey with the active Pester state that's already running". It doesn't seem like your tests should affect the Pester state at all since `Import-GherkinFeature`, while accepting a Pester State argument, doesn't even use it and you haven't even provided it.

    I do like how the "test data" (e.g. the feature file) you provided is inline with the test code, which is not a "mystery guest" like the other tests where this data is stored in external example files in the `..\Examples` directory. In fact, if what I suggest below works, I may end up using this style of "inline test data" for my future Pester Gherkin tests because I like it so much better! So, I think you could use a pattern very similar to the `Describe 'Invoke-Gherkin'` test in this case:

    ```powershell
    Describe "Get-StepParameters " {
        Context "Converts data in feature file steps" {
            $It "Should process a single-column table correctly" {
                # resolve the full name to the temporary feature file because gherkin doesn't support PSDrive paths
                $testDrive = (Get-PSDrive -Name "TestDrive").Root
                $featureFile = Join-Path -Path $testDrive -ChildPath "singlecolumn.feature"

                # write the temporary feature file that we're going to parse
                Set-Content -Path $featureFile -Value @'
    Feature: Gherkin integration test
    Scenario: The test data should be converted properly
        Given the test data
            | PropertyName |
            | Property1    |
            | Property2    |
            | Property3    |
    '@;

                # Calling this in a job so we don't monkey with the active pester state that's already running
                $job = Start-Job -ArgumentList $scriptRoot -ScriptBlock {
                    param ($scriptRoot)
                    Get-Module Pester | Remove-Module -Force
                    Import-Module $scriptRoot\Pester.psd1 -Force

                    $ActualDataTable = InModuleScope "Pester" {
                        # parse the feature file to extract the scenario data
                        $Feature, $Background, $Scenarios = Import-GherkinFeature -Path $featureFile

                        $NamedArguments, $Parameters = Get-StepParameters -Step $Scenarios.Steps[0] -CommandName "the test data";
                        $NamedArguments.Table
                    }
                    $ActualDataTable
                }

                $actualTable = $job | Wait-Job | Receive-Job
                Remove-Job $job

                # there must be an easier way to compare an array of hashtables?
                $expectedTable = @(
                    @{ "PropertyName" = "Property1" },
                    @{ "PropertyName" = "Property2" },
                    @{ "PropertyName" = "Property3" }
                );

                $actualTable.Length | Should -Be $expectedTable.Length;
                for( $i = 0; $i -lt $expectedTable.Length; $i++ ) {
                    $expectedTable[$i].Keys.Length | Should -Be $actualTable[$i].Keys.Length;
                    foreach( $key in $expectedTable[$i].Keys ) {
                        $key | Should -BeIn $actualTable[$i].Keys;
                        $actualTable[$i][$key] | Should -Be $expectedTable[$i][$key];
                    }
                }
            }

            # Rinse, dry, repeat for multi-column table example
        }
    }
    ```

Perhaps this would get you "unstuck"?